server/models/meshery_controllers.go

```diff
@@ -223,46 +231,46 @@ func (mch *MesheryControllersHelper) DeployUndeployedOperators(ot *OperatorTrack
 	if ot.DisableOperator { //Return true everytime so that operators stay in undeployed state across all contexts
 		return mch
 	}
-	go func(mch *MesheryControllersHelper) {
-		mch.mu.Lock()
-		defer mch.mu.Unlock()
-		for ctxID, ctrlHandler := range mch.ctxControllerHandlersMap {
-			if oprStatus, ok := mch.ctxOperatorStatusMap[ctxID]; ok {
-
-				if oprStatus == controllers.NotDeployed {
-
-					err := ctrlHandler[MesheryOperator].Deploy(false)
-					if err != nil {
-						mch.log.Error(err)
-					}
+	// go func(mch *MesheryControllersHelper) {
+	
+	if mch.ctxOperatorStatus == controllers.NotDeployed {
+		if mch.ctxControllerHandlers != nil {
+			operatorHandler, ok := mch.ctxControllerHandlers[MesheryOperator]
+			if ok {
+				err := operatorHandler.Deploy(false)
+
+				if err != nil {
+					mch.log.Error(err)
 				}
 			}
 		}
-	}(mch)
+	}
+
+	// }(mch)
 
 	return mch
 }
 
 func (mch *MesheryControllersHelper) UndeployDeployedOperators(ot *OperatorTracker) *MesheryControllersHelper {
-	go func(mch *MesheryControllersHelper) {
+	// go func(mch *MesheryControllersHelper) {
 
-		mch.mu.Lock()
-		defer mch.mu.Unlock()
-		for ctxID, ctrlHandler := range mch.ctxControllerHandlersMap {
+	oprStatus := mch.ctxOperatorStatus
 
-			if oprStatus, ok := mch.ctxOperatorStatusMap[ctxID]; ok {
+	if oprStatus != controllers.Undeployed {
```

Comment:
Or `controllers.Disabled`?

Reply:
Undeployed is one of the state that we have agreed upon, so I think it's better to use that.